broker-daemon/state-machines/plugins/state-machine-events.js

```diff
@@ -0,0 +1,69 @@
+const EventEmitter = require('events')
+
+const StateMachinePlugin = require('./abstract')
+
+/**
+ * Events that should trigger the cleanup of all event handlers created
+ * by the StateMachineEvents plugin
+ *
+ * @constant
+ * @type {Array<String>}
+ * @default
+ */
+const EXIT_EVENTS = Object.freeze(['cancel', 'complete', 'reject'])
+
+/**
+ * @class Add event emittion to a state machine
+ */
+class StateMachineEvents extends StateMachinePlugin {
+  /**
+   * Setup for configuration of StateMachineEvents plugin
+   * @returns {StateMachineEvents}
+   */
+  constructor () {
+    super()
+
+    this.eventHandler = new EventEmitter()
+  }
+
+  /**
+   * Adds an event emition observer to all transition calls to allow for event passing
+   * to external workers (BlockOrderWorker)
+   */
+  get observers () {
+    const plugin = this
+
+    return {
+      /**
+       * @param {Object} lifecycle
+       * @returns {void}
+       */
+      onAfterTransition: function (lifecycle) {
+        // We check to see if an order exists, if it does then we are dealing with an
+        // OSM, or else we use the FSM
+        const blockOrderId = this.order ? this.order.blockOrderId : this.fill.blockOrderId
```

Comment:
the state machine should not have knowledge of the block order, and this plugin should not have knowledge of how the order and fill state machines are structured.

We shouldn't be emitting the block order id at all - the block order is itself listening to this event emitter, so the ID is superfluous. Emitting the lifecycle event is plenty specific enough.

Reply:
@treygriffith I totally agree. I've removed the need for an ID during event handling. When these events are registered with a state machine, the caller should be providing the information needed. 

This code no longer exists